### ISSUE TITLE
Environment Variable Parsing Breaks on Colon

### ISSUE THREAD

**Issue by IanKnighton** [google-github-actions/deploy-cloudrun#374]
### TL;DR

I'm passing an image name with a handful of environment variables. The variables in front of the `:` parse correctly, but the variables after the `:` do not. 

### Expected behavior

All environment variables should be parsed. 

### Observed behavior

Only the variables in front of the `:` are parsed. 

### Action YAML

```yaml
name: Build and Publish to Google Artifact Registry

on:
  push:
    paths:
      - dir/**
    branches:
      - main
  pull_request:
    paths:
        - dir/**
    branches:
      - main

jobs:
  build:
    permissions:
        contents: 'read'
        id-token: 'write'
    name: Build and Push Docker Image
    runs-on: ubuntu-latest
    env:
      GH_TOKEN: ${{ secrets.PRIVATE_GITHUB_PACKAGES_TOKEN }}
      BUILD_NUMBER: $(date +%y%m%d).$GITHUB_RUN_NUMBER
      PROJECT_ID: PROJECT_ID
      GAR_LOCATION: LOCATION
      SERVICE: SERVICE_NAME
      IMAGE_NAME: IMAGE_NAME 
      REGION: REGION
      WIF_PROVIDER: ${{ secrets.WIF_PROVIDER }}
      WIF_ACCOUNT: ${{ secrets.WIF_SERVICE_ACCOUNT }}
    defaults:
      run:
        shell: bash
        working-directory: ./dir/
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Authenticate with Google
        id: auth
        uses: 'google-github-actions/auth@v0'
        with:
          token_format: 'access_token'
          workload_identity_provider: ${{ env.WIF_PROVIDER }}
          service_account: ${{ env.WIF_ACCOUNT }}
      
      - name: Authenticate to Docker
        id: docker-auth
        uses: 'docker/login-action@v1'
        with:
          username: 'oauth2accesstoken'
          password: '${{ steps.auth.outputs.access_token }}'
          registry: '${{ env.GAR_LOCATION }}-docker.pkg.dev'

      - name: Build and Push Container
        run: |-
          docker build -f ./dockerfile -t "${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.SERVICE }}/${{ env.IMAGE_NAME }}:${{ env.BUILD_NUMBER }}" ./ --build-arg GH_TOKEN=${{ env.GH_TOKEN }}
          docker push "${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.SERVICE }}/${{ env.IMAGE_NAME }}:${{ env.BUILD_NUMBER }}"
      
      - name: Deploy to Cloud Run
        id: deploy
        uses: google-github-actions/deploy-cloudrun@v0
        with:
          service: ${{ env.SERVICE }}
          region: ${{ env.REGION }}
          image: ${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.SERVICE }}/${{ env.IMAGE_NAME }}:${{ env.BUILD_NUMBER }}
```

### Log output

```text
2022-07-28T17:16:41.3329435Z ##[group]Run google-github-actions/deploy-cloudrun@v0
2022-07-28T17:16:41.3329817Z with:
2022-07-28T17:16:41.3330069Z   service: SERVICE_NAME
2022-07-28T17:16:41.3330340Z   region: REGION
2022-07-28T17:16:41.3330788Z   image: LOCATION-docker.pkg.dev/PROJECT_ID/SERVICE_NAME/IMAGE_NAME:$(date +%y%m%d).$GITHUB_RUN_NUMBER
2022-07-28T17:16:41.3331229Z   no_traffic: false
2022-07-28T17:16:41.3331486Z env:
2022-07-28T17:16:41.3331921Z   GH_TOKEN: ***
2022-07-28T17:16:41.3332235Z   BUILD_NUMBER: $(date +%y%m%d).$GITHUB_RUN_NUMBER
2022-07-28T17:16:41.3332579Z   PROJECT_ID: PROJECT_ID
2022-07-28T17:16:41.3332890Z   GAR_LOCATION: LOCATION
2022-07-28T17:16:41.3333174Z   SERVICE: SERVICE_NAME
2022-07-28T17:16:41.3333441Z   IMAGE_NAME: IMAGE_NAME
2022-07-28T17:16:41.3333819Z   REGION: REGION
2022-07-28T17:16:41.3334263Z   WIF_PROVIDER: ***
2022-07-28T17:16:41.3334652Z   WIF_ACCOUNT: ***
2022-07-28T17:16:41.3335144Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/REPO/REPO/gha-creds-46212e65624a26b9.json
2022-07-28T17:16:41.3335877Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/REPO/REPO/gha-creds-46212e65624a26b9.json
2022-07-28T17:16:41.3336519Z   GOOGLE_GHA_CREDS_PATH: /home/runner/work/REPO/REPO/gha-creds-46212e65624a26b9.json
2022-07-28T17:16:41.3336996Z   CLOUDSDK_CORE_PROJECT: PROJECT_ID
2022-07-28T17:16:41.3337321Z   CLOUDSDK_PROJECT: PROJECT_ID
2022-07-28T17:16:41.3337639Z   GCLOUD_PROJECT: PROJECT_ID
2022-07-28T17:16:41.3337941Z   GCP_PROJECT: PROJECT_ID
2022-07-28T17:16:41.3338253Z   GOOGLE_CLOUD_PROJECT: PROJECT_ID
2022-07-28T17:16:41.3338594Z ##[endgroup]
2022-07-28T17:16:42.3036168Z [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/1a974684-7236-47ff-bed2-8ebda0154aa5 -f /home/runner/work/_temp/d7026923-4678-48ed-878f-02a3ec79438d
2022-07-28T17:17:00.2906245Z Running: gcloud run deploy cars --image LOCATION-docker.pkg.dev/PROJECT_ID/SERVICE_NAME/IMAGE_NAME:$(date +%y%m%d).$GITHUB_RUN_NUMBER --quiet --platform managed --region REGION --project PROJECT_ID --format json
2022-07-28T17:17:02.9280490Z ##[error]google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy cars --image LOCATION-docker.pkg.dev/PROJECT_ID/SERVICE_NAME/IMAGE_NAME:$(date +%y%m%d).$GITHUB_RUN_NUMBER --quiet --platform managed --region REGION --project PROJECT_ID --format json`: Deploying container to Cloud Run service [cars] in project [PROJECT_ID] region [REGION]
Deploying...
Creating Revision.........interrupted
Deployment failed
ERROR: (gcloud.run.deploy) Revision 'SERVICE_NAME-00003-dog' is not ready and cannot serve traffic.
```

### Additional information

This action is the only one that seems to have an issue processing those last two variables. I've changed them to everything except plain text and always get the raw string. I've tried adding/removing quotes and everything else I can think of.

**Comment by sethvargo** [comment]
Hi @IanKnighton 

Thank you for opening an issue. This has nothing to do with Cloud Run or the Google GitHub Actions. You can see this issue reproduces without Cloud Run:

```yaml
name: repro

on:
  push:

jobs:
  build:
    runs-on: ubuntu-latest
    env:
      FOO: $(date +%y%m%d).$GITHUB_RUN_NUMBER
      BAR: $(date +%y%m%d)
      BAZ: $GITHUB_RUN_NUMBER

    steps:
      - name: repro
        run: |-
          echo "FOO: ${FOO}"
          echo "BAR: ${BAR}"
          echo "BAZ: ${BAZ}"
          echo "env.FOO: ${{ env.FOO }}"
          echo "env.BAR: ${{ env.BAR }}"
          echo "env.BAZ: ${{ env.BAZ }}"
```

The output will be:

```text
FOO: $(date +%y%m%d).$GITHUB_RUN_NUMBER
BAR: $(date +%y%m%d)
BAZ: $GITHUB_RUN_NUMBER
env.FOO: 220728.2
env.BAR: 220728
env.BAZ: 2
```

We don't have control over whether interpolation happens; GitHub Actions controls that. I would recommend opening an issue with GitHub.
